src/devices/srsmith_pool_srs_2c_tx.c

```diff
@@ -0,0 +1,175 @@
+
+Data Layout:
+    PPPP WWWW S UUUU C B T PP
+
+- P: 32 bit preamble (0xaaaaaaaa; 7 or 8 bits shifted left for analysis)
+- W: 32 bit sync word (0xd391d391)
+- S: 8 bit size (so far I've only seen 0x07)
+- U: 32 bit unknown (I always see 0x01fffff5 here)
+- C: 8 bit pin code is located in the bottom nibble of this byte, inverted and reversed.
+- B: 8 bit contains the ID of the button that was pushed on the remote
+- T: 8 bit CRC-8, poly 1, init 1 from the 16 bytes that we don't know (U) until the button that was pressed (B)
+- P: 16 bit CRC-16, poly 0x8005, init 0xFFFF, of the packet from the size (S) until the CRC-8 (T)
+
+Format String:
+    PRE:32h SYNC: 32h SIZE: hh UNSURE:32h | UNSURE: 4b | PIN ~^4b |  BTN: hh | CRC-8: hh | CRC-16: hhhh
+
+Capture raw but often misshifted :| data. if you want to work with this data, throw it into bitbench and shift it over until you see A's instead of 5's
+    -f 915000000 -X n=SRSmith,m=FSK_PCM,s=100,l=100,r=4096,match=d391d391
+*/
+
+static int srsmith_pool_srs_2c_tx_callback(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+    data_t *data;
+
+    if (bitbuffer->num_rows != 1)
+        return DECODE_ABORT_EARLY;
+
+    if (bitbuffer->bits_per_row[0] != 144)
+        return DECODE_ABORT_LENGTH;
+
+    // this is heavily derived from Maverick XR30 BBQ Sensor
+    uint8_t b[18];
+
+    // check for correct preamble/sync (0xaaaaaad391d391)
+    if (bitbuffer->bb[0][0] == 0x55) {
+        bitbuffer_extract_bytes(bitbuffer, 0, 7, b, 18 * 8); // shift in case first bit was not received properly
+    }
+    else if (bitbuffer->bb[0][0] == 0xaa) {
+        bitbuffer_extract_bytes(bitbuffer, 0, 8, b, 18 * 8); // shift in case first bit was received properly
+    }
+    else {
+        return DECODE_ABORT_EARLY; // preamble/sync missing
+    }
+    if (b[0] != 0xaa || b[1] != 0xaa || b[2] != 0xaa || b[3] != 0xd3 || b[4] != 0x91 || b[5] != 0xd3 || b[6] != 0x91)
+        return DECODE_ABORT_EARLY; // preamble/sync missing
+
+    int totallength = bitbuffer->bits_per_row[0];
+    int packetsize  = (int)b[7];
+
+    // packet actually starts at b[8]
+    uint32_t unknown_field = ((uint32_t)b[8] << 24) | (b[9] << 16) | (b[10] << 8) | (b[11]); // not sure what these four bytes are.
+
+    // get xmitted pin (byte 12, inverted, reversed, bottom nibble)
+    uint8_t pin_container          = b[12];
+    uint8_t inverted_pin_container = ~pin_container;                   // invert the pin
+    uint8_t reversed_pin           = reverse8(inverted_pin_container); // reverse the bits
+
+    // convert just the first four bits to binary string
+    char pin_string[5] = {'\0'};
+    snprintf(pin_string, sizeof(pin_string), "%d%d%d%d",
+            (reversed_pin & 0x80 ? 1 : 0),
+            (reversed_pin & 0x40 ? 1 : 0),
+            (reversed_pin & 0x20 ? 1 : 0),
+            (reversed_pin & 0x10 ? 1 : 0));
+
+    // button that was pushed
+    uint8_t button_id = b[13];
+
+    // get label
+    char button_string[50] = {'\0'};
```

Comment:
should be `char *button_string;`
and replace all `snprintf(button_string, sizeof(button_string), "On/Off Channel 1");` with `button_string = "On/Off Channel 1";`